src/btree/bt_curnext.c

```diff
@@ -172,6 +174,16 @@ __cursor_var_append_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart, size_
             return (WT_NOTFOUND);
 
         __cursor_set_recno(cbt, WT_INSERT_RECNO(cbt->ins));
+
+        /*
+         * If an upper bound has been set ensure that the key is within the range, otherwise early
+         * exit.
+         */
+        if ((ret = __wt_btcur_bounds_early_exit(session, cbt, true, key_out_of_boundsp)) ==
```

Comment:
I believe we want this after the `restart_read` section, thus if we return a prepare conflict we don't accidentally skip checking the bound after restarting the read.